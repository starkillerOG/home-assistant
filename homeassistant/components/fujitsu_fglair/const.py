"""Constants for the Fujitsu HVAC (based on Ayla IOT) integration."""

from datetime import timedelta

from ayla_iot_unofficial.fujitsu_consts import (  # noqa: F401
    FGLAIR_APP_ID,
    FGLAIR_APP_SECRET,
)
from ayla_iot_unofficial.fujitsu_hvac import FanSpeed, OpMode, SwingMode

from homeassistant.components.climate import (
    FAN_AUTO,
    FAN_HIGH,
    FAN_LOW,
    FAN_MEDIUM,
    SWING_BOTH,
    SWING_HORIZONTAL,
    SWING_OFF,
    SWING_VERTICAL,
    HVACMode,
)

API_TIMEOUT = 10
API_REFRESH = timedelta(minutes=5)

DOMAIN = "fujitsu_fglair"

CONF_EUROPE = "is_europe"

HA_TO_FUJI_FAN = {
    FAN_LOW: FanSpeed.LOW,
    FAN_MEDIUM: FanSpeed.MEDIUM,
    FAN_HIGH: FanSpeed.HIGH,
    FAN_AUTO: FanSpeed.AUTO,
}
FUJI_TO_HA_FAN = {value: key for key, value in HA_TO_FUJI_FAN.items()}

HA_TO_FUJI_HVAC = {
    HVACMode.OFF: OpMode.OFF,
    HVACMode.HEAT: OpMode.HEAT,
    HVACMode.COOL: OpMode.COOL,
    HVACMode.HEAT_COOL: OpMode.AUTO,
    HVACMode.DRY: OpMode.DRY,
    HVACMode.FAN_ONLY: OpMode.FAN,
}
FUJI_TO_HA_HVAC = {value: key for key, value in HA_TO_FUJI_HVAC.items()}

HA_TO_FUJI_SWING = {
    SWING_OFF: SwingMode.OFF,
    SWING_VERTICAL: SwingMode.SWING_VERTICAL,
    SWING_HORIZONTAL: SwingMode.SWING_HORIZONTAL,
    SWING_BOTH: SwingMode.SWING_BOTH,
}
FUJI_TO_HA_SWING = {value: key for key, value in HA_TO_FUJI_SWING.items()}
