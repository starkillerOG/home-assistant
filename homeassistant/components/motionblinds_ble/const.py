"""Constants for the MotionBlinds BLE integration."""

from enum import StrEnum

ATTR_CALIBRATION = "calibration"
ATTR_CONNECTION_TYPE = "connection_type"
ATTR_SPEED = "speed"
ATTR_BATTERY = "battery"
ATTR_CONNECT = "connect"
ATTR_DISCONNECT = "disconnect"
ATTR_FAVORITE = "favorite"
ATTR_SIGNAL_STRENGTH = "signal_strength"

CONF_LOCAL_NAME = "local_name"
CONF_ADDRESS = "address"
CONF_MAC_CODE = "mac_code"
CONF_BLIND_TYPE = "blind_type"

DOMAIN = "motionblinds_ble"

ENTITY_NAME = "MotionBlind {mac_code}"

ERROR_ALREADY_CONFIGURED = "already_configured"
ERROR_COULD_NOT_FIND_MOTOR = "could_not_find_motor"
ERROR_INVALID_MAC_CODE = "invalid_mac_code"
ERROR_NO_BLUETOOTH_ADAPTER = "no_bluetooth_adapter"
ERROR_NO_DEVICES_FOUND = "no_devices_found"

EXCEPTION_NOT_CALIBRATED = (
    "{device_name} needs to be calibrated using the MotionBlinds BLE app before usage."
)

ICON_CALIBRATION = "mdi:tune"
ICON_SPEED = "mdi:run-fast"
ICON_CONNECT = "mdi:bluetooth"
ICON_DISCONNECT = "mdi:bluetooth-off"
ICON_FAVORITE = "mdi:star"
ICON_CONNECTION_TYPE = "mdi:bluetooth-connect"
ICON_VERTICAL_BLIND = "mdi:blinds-vertical-closed"

MANUFACTURER = "MotionBlinds - Coulisse"

SETTING_MAX_MOTOR_FEEDBACK_TIME = 2  # Seconds


class MotionCalibrationType(StrEnum):
    """Enum for the blind calibration type."""

    CALIBRATED = "calibrated"
    UNCALIBRATED = "uncalibrated"
    CALIBRATING = "calibrating"


class MotionBlindType(StrEnum):
    """Enum for the blind type."""

    ROLLER = "roller"
    HONEYCOMB = "honeycomb"
    ROMAN = "roman"
    VENETIAN = "venetian"
    VENETIAN_TILT_ONLY = "venetian_tilt_only"
    DOUBLE_ROLLER = "double_roller"
    CURTAIN = "curtain"
    VERTICAL = "vertical"
