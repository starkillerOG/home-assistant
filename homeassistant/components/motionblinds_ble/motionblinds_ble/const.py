"""Constants for MotionBlinds BLE."""

from enum import IntEnum, StrEnum

SETTING_MAX_CONNECT_ATTEMPTS = 5
SETTING_MAX_COMMAND_ATTEMPTS = 5
SETTING_DISCONNECT_TIME = 15  # Seconds
SETTING_CALIBRATION_DISCONNECT_TIME = 45  # Seconds
SETTING_NOTIFICATION_DELAY = 0.5  # Seconds

EXCEPTION_NO_END_POSITIONS = (
    "{device_name}'s end positions need to be set before usage of this command."
)
EXCEPTION_NO_FAVORITE_POSITION = (
    "{device_name}'s favorite position needs to be set before usage of this command."
)


class MotionService(StrEnum):
    """Enum for the BLE service."""

    CONTROL = "d973f2e0-b19e-11e2-9e96-0800200c9a66"


class MotionCharacteristic(StrEnum):
    """Enum for the BLE characteristic."""

    COMMAND = "d973f2e2-b19e-11e2-9e96-0800200c9a66"
    NOTIFICATION = "d973f2e1-b19e-11e2-9e96-0800200c9a66"


class MotionCommandType(StrEnum):
    """Enum for the command type."""

    OPEN = "03020301"
    CLOSE = "03020302"
    STOP = "03020303"
    OPEN_TILT = "03020309"
    CLOSE_TILT = "0302030a"
    FAVORITE = "03020306"
    PERCENT = "05020440"
    ANGLE = "05020420"
    SPEED = "0403010a"
    SET_KEY = "02c001"
    STATUS_QUERY = "03050f02"
    USER_QUERY = "02c005"
    POINT_SET_QUERY = "03050120"


class MotionNotificationType(StrEnum):
    """Enum for the notification type."""

    POSITION = "07040402"
    STATUS = "12040f02"


class MotionConnectionType(StrEnum):
    """Enum for the connection type."""

    CONNECTED = "connected"
    CONNECTING = "connecting"
    DISCONNECTED = "disconnected"
    DISCONNECTING = "disconnecting"


class MotionRunningType(StrEnum):
    """Enum for the blind running type."""

    OPENING = "opening"
    CLOSING = "closing"
    STILL = "still"
    UNKNOWN = "unknown"


class MotionSpeedLevel(IntEnum):
    """Enum for the speed level."""

    LOW = 1
    MEDIUM = 2
    HIGH = 3
