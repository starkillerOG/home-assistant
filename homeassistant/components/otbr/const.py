"""Constants for the Open Thread Border Router integration."""

from __future__ import annotations

from typing import TYPE_CHECKING

from homeassistant.util.hass_dict import HassKey

if TYPE_CHECKING:
    from .util import OTBRData

DOMAIN = "otbr"
DATA_OTBR: HassKey[OTBRData] = HassKey(DOMAIN)

DEFAULT_CHANNEL = 15
