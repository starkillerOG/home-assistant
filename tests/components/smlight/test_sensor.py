"""Tests for the SMLIGHT sensor platform."""

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr, entity_registry as er

from .conftest import setup_integration

from tests.common import MockConfigEntry, snapshot_platform

pytestmark = [
    pytest.mark.usefixtures(
        "mock_smlight_client",
    )
]


@pytest.fixture
def platforms() -> Platform | list[Platform]:
    """Platforms, which should be loaded during the test."""
    return Platform.SENSOR


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_sensors(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    entity_registry: er.EntityRegistry,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the SMLIGHT sensors."""
    entry = await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)


async def test_disabled_by_default_sensors(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_config_entry: MockConfigEntry,
) -> None:
    """Test the disabled by default SMLIGHT sensors."""
    await setup_integration(hass, mock_config_entry)

    for sensor in ("ram_usage", "filesystem_usage"):
        assert not hass.states.get(f"sensor.mock_title_{sensor}")

        assert (entry := entity_registry.async_get(f"sensor.mock_title_{sensor}"))
        assert entry.disabled
        assert entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION
