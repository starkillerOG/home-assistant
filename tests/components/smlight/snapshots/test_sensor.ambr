# serializer version: 1
# name: test_sensors[sensor.mock_title_core_chip_temp-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.mock_title_core_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Core chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'core_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_core_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.mock_title_core_chip_temp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'Mock Title Core chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.mock_title_core_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '35.0',
  })
# ---
# name: test_sensors[sensor.mock_title_filesystem_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.mock_title_filesystem_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'Filesystem usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'fs_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_fs_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.mock_title_filesystem_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'Mock Title Filesystem usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.mock_title_filesystem_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '188',
  })
# ---
# name: test_sensors[sensor.mock_title_ram_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.mock_title_ram_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'RAM usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ram_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_ram_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.mock_title_ram_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'Mock Title RAM usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.mock_title_ram_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '99',
  })
# ---
# name: test_sensors[sensor.mock_title_zigbee_chip_temp-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.mock_title_zigbee_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Zigbee chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'zigbee_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_zigbee_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.mock_title_zigbee_chip_temp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'Mock Title Zigbee chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.mock_title_zigbee_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '32.7',
  })
# ---
# name: test_sensors[sensor.slzb_06_core_chip_temp-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_core_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Core chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'core_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_core_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_core_chip_temp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'slzb-06 Core chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_core_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '35.0',
  })
# ---
# name: test_sensors[sensor.slzb_06_core_chip_temp]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'slzb-06 Core chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_core_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '35.0',
  })
# ---
# name: test_sensors[sensor.slzb_06_core_chip_temp].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_core_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Core chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'core_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_core_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_core_chip_temp].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'http://slzb-06.local',
    'connections': set({
      tuple(
        'mac',
        'aa:bb:cc:dd:ee:ff',
      ),
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'SMLIGHT',
    'model': 'SLZB-06p7',
    'model_id': None,
    'name': 'slzb-06',
    'name_by_user': None,
    'primary_config_entry': <ANY>,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': 'core: v2.3.1.dev / zigbee: -1',
    'via_device_id': None,
  })
# ---
# name: test_sensors[sensor.slzb_06_filesystem_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_filesystem_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'Filesystem usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'fs_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_fs_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_filesystem_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'slzb-06 Filesystem usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_filesystem_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '188',
  })
# ---
# name: test_sensors[sensor.slzb_06_filesystem_usage]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'slzb-06 Filesystem usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_filesystem_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '188',
  })
# ---
# name: test_sensors[sensor.slzb_06_filesystem_usage].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_filesystem_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'Filesystem usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'fs_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_fs_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_filesystem_usage].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'http://slzb-06.local',
    'connections': set({
      tuple(
        'mac',
        'aa:bb:cc:dd:ee:ff',
      ),
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'SMLIGHT',
    'model': 'SLZB-06p7',
    'model_id': None,
    'name': 'slzb-06',
    'name_by_user': None,
    'primary_config_entry': <ANY>,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': 'core: v2.3.1.dev / zigbee: -1',
    'via_device_id': None,
  })
# ---
# name: test_sensors[sensor.slzb_06_ram_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_ram_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'RAM usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ram_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_ram_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_ram_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'slzb-06 RAM usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_ram_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '99',
  })
# ---
# name: test_sensors[sensor.slzb_06_ram_usage]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'data_size',
      'friendly_name': 'slzb-06 RAM usage',
      'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_ram_usage',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '99',
  })
# ---
# name: test_sensors[sensor.slzb_06_ram_usage].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_ram_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DATA_SIZE: 'data_size'>,
    'original_icon': None,
    'original_name': 'RAM usage',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ram_usage',
    'unique_id': 'aa:bb:cc:dd:ee:ff_ram_usage',
    'unit_of_measurement': <UnitOfInformation.KILOBYTES: 'kB'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_ram_usage].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'http://slzb-06.local',
    'connections': set({
      tuple(
        'mac',
        'aa:bb:cc:dd:ee:ff',
      ),
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'SMLIGHT',
    'model': 'SLZB-06p7',
    'model_id': None,
    'name': 'slzb-06',
    'name_by_user': None,
    'primary_config_entry': <ANY>,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': 'core: v2.3.1.dev / zigbee: -1',
    'via_device_id': None,
  })
# ---
# name: test_sensors[sensor.slzb_06_zigbee_chip_temp-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_zigbee_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Zigbee chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'zigbee_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_zigbee_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_zigbee_chip_temp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'slzb-06 Zigbee chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_zigbee_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '32.7',
  })
# ---
# name: test_sensors[sensor.slzb_06_zigbee_chip_temp]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'slzb-06 Zigbee chip temp',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.slzb_06_zigbee_chip_temp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': '32.7',
  })
# ---
# name: test_sensors[sensor.slzb_06_zigbee_chip_temp].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.slzb_06_zigbee_chip_temp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
      'sensor': dict({
        'suggested_display_precision': 1,
      }),
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': None,
    'original_name': 'Zigbee chip temp',
    'platform': 'smlight',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'zigbee_temperature',
    'unique_id': 'aa:bb:cc:dd:ee:ff_zigbee_temperature',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_sensors[sensor.slzb_06_zigbee_chip_temp].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': 'http://slzb-06.local',
    'connections': set({
      tuple(
        'mac',
        'aa:bb:cc:dd:ee:ff',
      ),
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
    }),
    'is_new': False,
    'labels': set({
    }),
    'manufacturer': 'SMLIGHT',
    'model': 'SLZB-06p7',
    'model_id': None,
    'name': 'slzb-06',
    'name_by_user': None,
    'primary_config_entry': <ANY>,
    'serial_number': None,
    'suggested_area': None,
    'sw_version': 'core: v2.3.1.dev / zigbee: -1',
    'via_device_id': None,
  })
# ---
